code/datums/components/weed_food.dm

```diff
@@ -193,6 +193,8 @@
 			return FALSE
 	if(!parent_turf?.weeds)
 		return FALSE
+	if(SEND_SIGNAL(parent_mob, COMSIG_ATTEMPT_MOB_PULL) & COMPONENT_CANCEL_MOB_PULL)
```

Comment:
signals should generally have one point of origin ( https://hackmd.io/@tgstation/SignalsComponentsElements#Signals-How-they-work ), why not just send additional signals here that're registered in /proc/get_hung()?

Reply:
Look again at how that signal is used, its usage is to test whether the item is pullable; not an event that something occured like most signals like you're describing.

Reply:
Basically this is an abstract way asking any components that have registered COMSIG_ATTEMPT_MOB_PULL to deny the request. Otherwise pred hanging needs to remove this signal registration, and guard against its mobs in some other way.

Reply:
i don’t really like the pattern but it’s fine you didn’t start it